src/browser-polyfill.js

```diff
@@ -76,18 +76,30 @@ if (typeof browser === "undefined") {
      *        The promise's resolution function.
      * @param {function} promise.rejection
      *        The promise's rejection function.
+     * @param {object} metadata
+     *        Metadata about the wrapped method which has created the callback.
+     * @param {integer} metadata.maxResolvedArgs
+     *        The maximum number of arguments which may be passed to the
+     *        callback created by the wrapped async function.
      *
      * @returns {function}
      *        The generated callback function.
      */
-    const makeCallback = promise => {
+    const makeCallback = (promise, metadata) => {
       return (...callbackArgs) => {
         if (chrome.runtime.lastError) {
           promise.reject(chrome.runtime.lastError);
-        } else if (callbackArgs.length === 1) {
-          promise.resolve(callbackArgs[0]);
         } else {
-          promise.resolve(callbackArgs);
+          if ("maxResolvedArgs" in metadata) {
+            // Some chrome APIs provides to their callback more arguments than expected.
+            callbackArgs = callbackArgs.slice(0, metadata.maxResolvedArgs);
+          }
```

Comment:
There is no need to be so generic. Since the problem only occurs when with 1 vs not-one parameter, I suggest to throw away "maxResolvedArgs" and use `"singleCallbackParameter": true` or something similar. This could also be used to fix API definitions for callbacks that don't expect any parameter. Based on the code below, it is likely that callbacks without parameters resolve to an empty array.